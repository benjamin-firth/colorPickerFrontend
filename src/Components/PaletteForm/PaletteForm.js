import React, { useState, useEffect } from 'react';
import './PaletteForm.scss';
import PropTypes from 'prop-types';
import { useDispatch, useSelector } from 'react-redux';
import { addPalette } from '../../actions/index';

function PaletteForm({ currentPalette }) {

  const [paletteName, setPaletteName] = useState('');

  const handleSubmit = () => {
    const chosenPalette = {
      test: 'test'
    }
  }

  return (
    <form className='palette-form' onSubmit={handleSubmit}>
      <select >
        <option id={'test'} value="project1">Project 1</option>
      </select>
      <input 
        type='text' 
        onChange={(e) => setPaletteName(e.target.value)}/>
      <button 
        type='button'  
        className='submit-palette'>Save Palette
      </button>
    </form>
  );
}

export default PaletteForm;
