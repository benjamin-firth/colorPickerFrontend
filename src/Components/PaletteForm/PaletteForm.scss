.submit-palette-button {
  box-shadow: 3px 3px;
  background-color: #CDCDCD;
}
