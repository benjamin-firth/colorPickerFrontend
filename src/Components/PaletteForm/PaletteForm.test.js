import React from 'react';
import { render } from '@testing-library/react';
import PaletteForm from './PaletteForm';
import { shallow } from 'enzyme';


let mockCurrentPalette = {
  name: 'test',
  project_id: 2,
  color1: 'FFF',
  color2: 'FFF',
  color3: 'FFF',
  color4: 'FFF',
  color5: 'FFF',
  id: 3
};

jest.mock("react-redux", () => ({
  useSelector: () => [{name: "Test", id: 1}],
  useDispatch: () => jest.fn()
}));

describe('PaletteForm', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<PaletteForm currentPalette={mockCurrentPalette}/>)
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
