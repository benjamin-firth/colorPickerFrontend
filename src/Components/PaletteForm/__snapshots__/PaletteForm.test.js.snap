// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaletteForm should match the snapshot 1`] = `
<form
  className="palette-form"
>
  <select
    onChange={[Function]}
    value="1"
  >
    <option
      id={1}
      value={1}
    >
      Test
    </option>
  </select>
  <input
    onChange={[Function]}
    type="text"
  />
  <button
    className="submit-palette-button"
    onClick={[Function]}
    type="button"
  >
    Save Palette
  </button>
</form>
`;
